File: back_end_doctor_app/src/main/java/com/app/doctor/s_apointment/service/SlotService.java
package com.app.doctor.s_apointment.service;

import java.time.LocalDate;
import java.util.List;

import com.app.doctor.s_apointment.beans.Slot;

public interface SlotService {
	   Slot addSlot(Slot slot);

	    List<Slot> getAll();

		void deleteSlotById(int slotid);

		List<Slot> unCateredSlotById(String doctor_id);

		List<Slot> cateredSlotByDate(String doctor_id, LocalDate date);

		String bookSlot(String slotId, String PatientId);

		String changeSlotStaus(String slotId);

		List<Slot> getBookedSlot(String doctor_id);
}

File: front_end_doctor_app/src/service/SlotService.js
import axios from 'axios';
const BASE_URL="http://localhost:8080/slot/";
class SlotService{

    
     bookSlot( slotId,patientId){
        console.log("slot : "+slotId+" pid :"+patientId)
        return axios.put(BASE_URL+"book/"+slotId+"/"+patientId);
    }

    addSlot(slot){
        
        return axios.post(BASE_URL+"add/", slot);
    }

    deleteSlotById( slotId){
        
        return axios.delete(BASE_URL+"delete/"+slotId);
    }

    unCateredSlotById( doctor_id){
        
        return axios.get(BASE_URL+"uncatered/"+doctor_id);
    }

    cateredSlotByDate( doctor_id,date){
        
        return axios.get(BASE_URL+"catered/"+doctor_id+"/"+date);
    }

    getBookedSlot( doctor_id){
        
        return axios.get(BASE_URL+"incomp/"+doctor_id);
    }

}
export default new SlotService();

